2015-Fall/lrusnac/CalculatorV2/CalculatorV2/CalculatorViewController.swift

```diff
@@ -0,0 +1,119 @@
+//  Copyright (c) 2015 Leonid Rusnac. All rights reserved.
+//
+
+import UIKit
+
+class CalculatorViewController: UIViewController {
+
+    @IBOutlet weak var display: UILabel!
+    
+    var calculatorModel = CalculatorModel()
+    var userIsInTheMiddleOfTypingANumber = false
+    var pointWasInsertedInThisNumber = false
+    
+    @IBAction func appendDigit(sender: UIButton) {
+        if let digit = sender.currentTitle {
+            if userIsInTheMiddleOfTypingANumber {
+                display.text = display.text! + digit
+            } else {
+                display.text = digit
+                userIsInTheMiddleOfTypingANumber = true
+            }
+        }
+    }
+
+    @IBAction func clear() {
+        calculatorModel.clear()
+        display.text = "\(0)"
+        userIsInTheMiddleOfTypingANumber = false
+        pointWasInsertedInThisNumber = false
+    }
+    
+    @IBAction func backspace() {
+        // I let the user to backspace only if it is in middle of typing
+        if userIsInTheMiddleOfTypingANumber {
+            if let displayText = display.text {
+                let newDisplay = dropLast(displayText)
+                if newDisplay.isEmpty {
+                    userIsInTheMiddleOfTypingANumber = false
+                    pointWasInsertedInThisNumber = false
+                    display.text = "0"
+                } else {
+                    if let pointIndex = newDisplay.rangeOfString(".") {
+                    } else {
+                        pointWasInsertedInThisNumber = false
+                    }
+                    display.text = newDisplay
+                }
+            }
+        }
+    }
+    
+    @IBAction func enter() {
+        if userIsInTheMiddleOfTypingANumber {
+            //push number and evaluate the stack
+            if let val = displayValue {
+                calculatorModel.pushNumber(val)
+                if let res = calculatorModel.evaluate() {
+                    displayValue = res
+                }
+                userIsInTheMiddleOfTypingANumber = false
+                pointWasInsertedInThisNumber = false
+            }
+        }
+    }
+    
+    @IBAction func point() {
+        if !pointWasInsertedInThisNumber {
+            if userIsInTheMiddleOfTypingANumber {
+                display.text = display.text! + "."
+            } else {
+                display.text = "0."
+                userIsInTheMiddleOfTypingANumber = true
+            }
+            pointWasInsertedInThisNumber = true
+        }
+    }
+    
+    @IBAction func addOperator(sender: UIButton) {
+        enter()
+        if let op = sender.currentTitle {
+            calculatorModel.pushOpNotNumber(op)
+            userIsInTheMiddleOfTypingANumber = false
+            pointWasInsertedInThisNumber = false
+        }
+        
+        if let res = calculatorModel.evaluate() {
+            displayValue = res
+        }
+    }
+    
+    @IBAction func addConstant(sender: UIButton) {
```

Comment:
That is a very valid point. It is not unlikely that it might be nice to do different things for these two different actions. In fact, I think it is a very good idea to have different methods for different things.

However, since both methods have the exact same method bodies (the code in them) you should extract this code into a common helper method that both of these actions utilise.